README.md

```diff
@@ -1,99 +1,98 @@
+the name of the entry itself. For example, for entry `foo.service.conf` it is expected that
+`foo.service.conf.j2` Jinja template exists and resulting dropin file will extend `foo.service`
+unit file.
 
-on the *control node* before using the role.
+### systemd_started_units
 
-## Role Variables
+List of unit names that shall be started via systemd.
+
+### systemd_stopped_units
+
+List of unit names that shall be stopped via systemd.
+
+### systemd_restarted_units
+
+List of unit names that shall be restarted via systemd.
 
-A description of all input variables (i.e. variables that are defined in
-`defaults/main.yml`) for the role should go here as these form an API of the
-role.  Each variable should have its own section e.g.
+### systemd_reloaded_units
 
-### template_foo
+List of unit names that shall be reloaded via systemd.
 
-This variable is required.  It is a string that lists the foo of the role.
-There is no default value.
+### systemd_enabled_units
 
-### template_bar
+List of unit files that shall be enabled via systemd.
 
-This variable is optional.  It is a boolean that tells the role to disable bar.
-The default value is `true`.
+### systemd_disabled_units
 
-Variables that are not intended as input, like variables defined in
-`vars/main.yml`, variables that are read from other roles and/or the global
-scope (ie. hostvars, group vars, etc.) can be also mentioned here but keep in
-mind that as these are probably not part of the role API they may change during
-the lifetime.
+List of unit files that shall be disabled via systemd.
+
+### systemd_masked_units
+
+List of unit files that shall be masked via systemd.
+
+### systemd_unmasked_units
+
+List of unit files that shall be unmasked via systemd.
 
 Example of setting the variables:
 
 ```yaml
-template_foo: "oof"
-template_bar: false
+systemd_unit_files: ['foo.service', 'bar.service']
+systemd_dropins: ['cups.service.conf', 'avahi-daemon.service.conf']
+systemd_started_units: ['foo.service', 'bar.service']
+systemd_enabled_units: ['foo.service', 'bar.service']
 ```
 
 ## Variables Exported by the Role
 
-This section is optional.  Some roles may export variables for playbooks to
-use later.  These are analogous to "return values" in Ansible modules.  For
-example, if a role performs some action that will require a system reboot, but
-the user wants to defer the reboot, the role might set a variable like
-`template_reboot_needed: true` that the playbook can use to reboot at a more
-convenient time.
+### `systemd_units`
 
-Example:
-
-### template_reboot_needed
-
-Default `false` - if `true`, this means a reboot is needed to apply the changes
-made by the role
+Variable shall contain a list of dictionaries where each entry describes state of one systemd unit
+present on the managed host. Note that this variable is not exported automatically. To export it 
+you need to invoke `systemd_units` module without any arguments.
```

Comment:
Modules provided by system roles are not callable directly - the only interface we support is the role. Calling the *role* with no arguments, or with some specified arguments, should return `systemd_units`.  For example, see https://github.com/linux-system-roles/firewall#gathering-firewall-ansible-facts

```yaml
- name: Get systemd unit facts
  include_role:
    name: linux-system-roles.systemd
  systemd_unit_files:
```
Then in tasks/main.yml differentiate between `when: systemd_unit_files is none` vs `when: system_unit_files is not none`

Reply:
it depends on how the variable is set - I think variables set with `set_fact` and `register` are in the global namespace

Reply:
I am not sure I understand, I guess we should always return systemd_units facts. If you call role w/o arguments you get current state of the system, on the other hand if some arguments are passed is then facts might already account for changes done by role in earlier tasks (i.e. some service was started).

Reply:
I've used `set_fact:`